hypersync-format/src/types/quantity.rs

```diff
@@ -221,8 +228,9 @@ mod tests {
 
     #[test]
     fn test_serde_zero() {
+        use serde_test::Configure;
         assert_eq!(Quantity::default(), Quantity::from(vec![0]));
-        assert_tokens(&Quantity::default(), &[Token::Str("0x0")]);
+        assert_tokens(&Quantity::default().readable(), &[Token::Str("0x0")]);
```

Comment:
I changed Quantity to accept both hex strings and plain integers for JSON (to support chains like Sonic). The .readable() method tells our test harness which behavior to test - JSON format (where integers are allowed) vs binary format (where only strings work)